src/configuration-requestor.ts

```diff
@@ -25,14 +26,15 @@ export default class ConfigurationRequestor {
       this.banditVariationConfigurationStore,
       this.banditModelConfigurationStore,
     );
+    this.configurationCopy = this.configuration.copy();
```

Comment:
This could be large; I wonder if we do the copying just in time when `getConfiguration()` is called. We could still check `this.configurationCopy` so that we only do it once.